dashboard/test/controllers/aichat_controller_test.rb

```diff
@@ -244,4 +245,33 @@ class AichatControllerTest < ActionController::TestCase
     assert_equal json_response['executionStatus'], execution_status
     assert_equal json_response['response'], response
   end
+
+  # user_has_aichat_access tests
+  test 'GET user_has_aichat_access returns false for unauthorized teacher' do
+    sign_in(create(:teacher))
+    get :user_has_aichat_access
+    assert_response :success
+    assert_equal json_response, false
+  end
+
+  test 'GET user_has_aichat_access returns true for authorized teacher' do
+    sign_in(@authorized_teacher1)
+    get :user_has_aichat_access
+    assert_response :success
+    assert_equal json_response, true
+  end
+
+  test 'GET user_has_aichat_access returns false for unauthorized student' do
+    sign_in(create(:student))
+    get :user_has_aichat_access
+    assert_response :success
+    assert_equal json_response, false
+  end
+
+  test 'GET user_has_aichat_access returns true for student of authorized teacher' do
+    sign_in(@authorized_student1)
+    get :user_has_aichat_access
+    assert_response :success
+    assert_equal json_response, true
+  end
```

Comment:
What about if there's no signed-in user? I think `current_user&.has_aichat_access?` would return `nil` which maybe translates to `null` in json? Would we rather return `false` result in that case? Adding a test for no signed-in user would help explicitly define what we expect in that case

Reply:
If user is not signed in, then endpoint/action is forbidden (See `ability.rb`). I can add a test for this!

Reply:
> Because it's inside a block if user.persisted??

Yes - that's correct. Just getting used to permissions myself - thanks!